From 41b1f9f0a066da30f799a2a99a313fd558ce68f5 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Tue, 15 Jul 2025 17:32:03 -0300
Subject: [PATCH] Link g++-mapper-server with -lnetwork on Haiku

---
 c++tools/Makefile.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/c++tools/Makefile.in b/c++tools/Makefile.in
index 5acc6d681..3abff97dc 100644
--- a/c++tools/Makefile.in
+++ b/c++tools/Makefile.in
@@ -37,6 +37,10 @@ exeext := @EXEEXT@
 LIBIBERTY := ../libiberty/libiberty.a
 VERSION.O := ../gcc/version.o
 
+ifneq (,$(findstring haiku,$(target_noncanonical)))
+    LDFLAGS += -lnetwork
+endif
+
 all::
 
 mostlyclean::
-- 
2.49.0

