From d70554680d1327e2322fe3097d80f214f9755fa3 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Tue, 15 Jul 2025 17:32:03 -0300
Subject: [PATCH] Link g++-mapper-server with -lnetwork on Haiku

---
 c++tools/Makefile.in | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/c++tools/Makefile.in b/c++tools/Makefile.in
index 5acc6d681..117b168bf 100644
--- a/c++tools/Makefile.in
+++ b/c++tools/Makefile.in
@@ -32,11 +32,23 @@ CXXFLAGS := @CXXFLAGS@
 PICFLAG := @PICFLAG@
 LD_PICFLAG := @LD_PICFLAG@
 CXXOPTS := $(CXXFLAGS) $(PICFLAG) -fno-exceptions -fno-rtti
-LDFLAGS := @LDFLAGS@
+HAIKU_LDFLAGS =
+
+ifneq (,$(findstring haiku,$(host)))
+    HAIKU_LDFLAGS += -lnetwork
+endif
+
+LDFLAGS := $(HAIKU_LDFLAGS) @LDFLAGS@
 exeext := @EXEEXT@
 LIBIBERTY := ../libiberty/libiberty.a
 VERSION.O := ../gcc/version.o
 
+HAIKU_LDFLAGS = 
+
+ifneq (,$(findstring haiku,$(host)))
+        HAIKU_LDFLAGS += -lnetwork
+endif
+
 all::
 
 mostlyclean::
@@ -98,7 +110,7 @@ MAPPER.O := server.o resolver.o
 CODYLIB = ../libcody/libcody.a
 CXXINC += -I$(srcdir)/../libcody -I$(srcdir)/../include -I$(srcdir)/../gcc -I. -I../gcc
 g++-mapper-server$(exeext): $(MAPPER.O) $(CODYLIB)
-	+$(CXX) $(LDFLAGS) $(PICFLAG) $(LD_PICFLAG) -o $@ $^ $(LIBIBERTY)
+	+$(CXX) $(LDFLAGS) $(HAIKU_LDFLAGS) $(PICFLAG) $(LD_PICFLAG) -o $@ $^ $(LIBIBERTY)
 
 # copy to gcc dir so tests there can run
 all::../gcc/g++-mapper-server$(exeext)
-- 
2.49.0

