From a059759f6ef36611d93c63cae1159e9bbe426d89 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Tue, 15 Jul 2025 17:32:03 -0300
Subject: [PATCH] Link g++-mapper-server with -lnetwork on Haiku

---
 c++tools/Makefile.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/c++tools/Makefile.in b/c++tools/Makefile.in
index 5acc6d681..38f94213c 100644
--- a/c++tools/Makefile.in
+++ b/c++tools/Makefile.in
@@ -37,6 +37,8 @@ exeext := @EXEEXT@
 LIBIBERTY := ../libiberty/libiberty.a
 VERSION.O := ../gcc/version.o
 
+LDFLAGS += -lnetwork
+
 all::
 
 mostlyclean::
-- 
2.49.0

